sinks/sinks_service.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+// Adapted for Orb project, modifications licensed under MPL v. 2.0:
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+package sinks
+
+import (
+	"context"
+	"fmt"
+	"github.com/mainflux/mainflux"
+	"github.com/ns1labs/orb/pkg/errors"
+	"github.com/ns1labs/orb/sinks/backend"
+)
+
+var (
+	ErrCreateSink                 = errors.New("failed to create Sink")
+	ErrConflictSink               = errors.New("entity already exists")
+	ErrUnsupportedContentTypeSink = errors.New("unsupported content type")
+)
+
+func (svc sinkService) CreateSink(ctx context.Context, token string, sink Sink) (Sink, error) {
+
+	mfOwnerID, err := svc.identify(token)
+	if err != nil {
+		return Sink{}, err
+	}
+
+	sink.MFOwnerID = mfOwnerID
+
+	if backend.HaveBackend(sink.Backend) {
+		err = backend.GetBackend(sink.Backend).Connect(sink.Config)
+		if err != nil {
+			sink.Status = "not connected"
+			sink.Error = fmt.Sprint(err)
+		} else {
+			sink.Status = "connected"
+			sink.Error = ""
+		}
+	}
+
+	id, err := svc.sinkRepo.Save(ctx, sink)
+	if err != nil {
+		return Sink{}, errors.Wrap(ErrCreateSink, err)
+	}
+	sink.ID = id
+	return sink, nil
+}
+
+func (svc sinkService) UpdateSink(ctx context.Context, token string, sink Sink) error {
+	skOwnerID, err := svc.identify(token)
+	if err != nil {
+		return err
+	}
+
+	sink.MFOwnerID = skOwnerID
+
+	if backend.HaveBackend(sink.Backend) {
```

Comment:
changing the backend type is not allowed, so this should not be needed. we should make sure that they can only update tags, description, and the other fields which are not read only